### DISCUSSION TITLE
Should I explore Logtalk, if I want to use Prolog in a wsh/wsf environment?

### DISCUSSION THREAD

**Discussion by GHExplorer** [LogtalkDotOrg/logtalk3#107] [Q&A]
Hi,

I tried using the tau-prolog Javascript module, over at http://tau-prolog.org/tau-prolog.js, in a .wsf file.  But it generates various expression errors.

I'd like to just play with the scripting environment, mixing VBScript & JScript, and calling prolog modules.

Is LogTalk appropriate, for this purpose?  Or is there some other JScript implementation that might be smoother?

My experience so far, is with Win/LPA-Prolog.

**Comment by pmoura** [comment]
I'm not familiar with `.wsf` files but the Logtalk distribution includes a `taulgt.sh` Bash shell script that start Logtalk with Tau Prolog that may be useful for you to play with and try some sample queries. Be aware, however, that due to its JavaScript implementation and the size of Logtalk that startup is slow.

Logtalk can call backend Prolog module predicates. That should not be an issue. Tau Prolog is the only JavaScript implementation of Prolog that's supported by Logtalk. There's also [ProscriptLS](https://github.com/lindseyspratt/proscriptls) but currently it lacks the necessary standards compliance that would allow it to run Logtalk.

I'm aware of Win/LPA-Prolog but this system is also not supported by Logtalk due to standards compliance issues.

**Reply by GHExplorer** [reply]
Thanks pmoura,

It sounds like it won't work for me.  I'm in the Windows environment, and the problem is that tao-prolog won't compile as a JScript module ; only as strict Javascript.


But the ProscriptLS example you provided, might be worth exploring.  It also leads to other implementations.

Yeah, Win-Prolog has a really nice external language interface.  I have a licensed copy, but I'd like to work with a Prolog implementation that I can freely share & distribute.

I'll look at the LogTalk distribution over the weekend, as well as the ProscriptLS-related links.

This is the .wsf code I'm working with, btw:

<package>
	<job id="test">
		<script language="JScript" src="tau-prolog.js" >
		</script>
		<script language="JScript">
			// first test of tau prolog in a wsf file
			var Var_x ;
			var_x = "what the...?" ;
			WScript.Echo( var_x ) ;
		</script>
	</job>
</package>	



Thanks,
